combatevents.lua

```diff
@@ -0,0 +1,116 @@
+local _, ns = ...
+local oUF = ns.oUF
+local Private = oUF.Private
+
+local argcheck = Private.argcheck
+local frame_metatable = Private.frame_metatable
+
+local CombatLogGetCurrentEventInfo = _G.CombatLogGetCurrentEventInfo
+
+local event_metatable = {
+	__call = function(funcs, ...)
+		for self, func in next, funcs do
+			if (self:IsVisible()) then
+				func(self, ...)
+			end
+		end
+	end,
+}
+
+local self_metatable = {
+	__call = function(funcs, self, ...)
+		for _, func in next, funcs do
+			func(self, ...)
+		end
+	end
+}
+
+local listener = CreateFrame('Frame')
+listener.activeEvents = 0
+
+local function filter(_, event, ...)
```

Comment:
I'd expand on the name of this function, processEvent perhaps?

Reply:
It actually filters the events and handles the processing to the registered listener. Event is here the combat log event, maybe I should rather rename this?